flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriter.java

```diff
@@ -137,8 +137,9 @@ boolean isDone() {
 
 	/**
 	 * Aborts the checkpoint and fails pending result for this checkpoint.
+	 * @param cleanup true if {@link #getWriteResult(long)} is not supposed to be called afterwards.
 	 */
-	void abort(long checkpointId, Throwable cause);
+	void abort(long checkpointId, Throwable cause, boolean cleanup);
```

Comment:
I guess the intend of cleanup is to avoid throwing an exception if `getWriteResult(long)` is called nonetheless, right?
So in which cases can that happen? Do we ensure cleanup in these cases (e.g., is abort then called twice?)?

Reply:
> I guess the intend of cleanup is to avoid throwing an exception if getWriteResult(long) is called nonetheless, right?

Yes.

> So in which cases can that happen?

`abort` with `cleanup=false` can be called when:
1. receiving abort notification via RPC
2. `CheckpointBarrierUnaligner.allBarriersReceivedFuture` is cancelled

> Do we ensure cleanup in these cases (e.g., is abort then called twice?)?

Yes, for each checkpoint either "regular" `abort` or `getWriteResult` is eventually called.